core/src/bson_type_info.rs

```diff
@@ -1,214 +1,426 @@
+        is_unsigned: None,
+        num_prec_radix: None,
     };
     pub const LONG: BsonTypeInfo = BsonTypeInfo {
         type_name: "long",
         sql_type: SqlDataType::EXT_BIG_INT,
-        searchable: true,
+        non_concise_type: SqlDataType::EXT_BIG_INT,
+        searchable: SQL_PRED_BASIC,
+        is_case_sensitive: false,
+        fixed_prec_scale: true,
         scale: Some(0),
         precision: Some(19),
         octet_length: Some(8),
         fixed_bytes_length: Some(8),
+        literal_prefix: None,
+        literal_suffix: None,
+        sql_code: None,
+        is_auto_unique_value: Some(false),
+        is_unsigned: Some(false),
+        num_prec_radix: Some(10),
     };
     pub const DECIMAL: BsonTypeInfo = BsonTypeInfo {
         type_name: "decimal",
         // TODO SQL-1068: Change to SqlDataType::DECIMAL
         sql_type: SqlDataType::UNKNOWN_TYPE,
-        searchable: true,
+        non_concise_type: SqlDataType::UNKNOWN_TYPE,
+        searchable: SQL_PRED_BASIC,
+        is_case_sensitive: false,
+        fixed_prec_scale: false,
         scale: None,
         precision: None,          // TODO SQL-1068: Some(34),
         octet_length: None,       // TODO SQL-1068: Some(16),
         fixed_bytes_length: None, // TODO SQL-1068: Some(16),
+        literal_prefix: None,
+        literal_suffix: None,
+        sql_code: None,
+        is_auto_unique_value: Some(false),
+        is_unsigned: Some(false),
+        num_prec_radix: None,
     };
     pub const MINKEY: BsonTypeInfo = BsonTypeInfo {
         type_name: "minKey",
         sql_type: SqlDataType::UNKNOWN_TYPE,
-        searchable: true,
+        non_concise_type: SqlDataType::UNKNOWN_TYPE,
+        searchable: SQL_PRED_BASIC,
+        is_case_sensitive: false,
+        fixed_prec_scale: false,
         scale: None,
         precision: None,
         octet_length: None,
         fixed_bytes_length: None,
+        literal_prefix: None,
+        literal_suffix: None,
+        sql_code: None,
+        is_auto_unique_value: None,
+        is_unsigned: None,
+        num_prec_radix: None,
     };
     pub const MAXKEY: BsonTypeInfo = BsonTypeInfo {
         type_name: "maxKey",
         sql_type: SqlDataType::UNKNOWN_TYPE,
-        searchable: true,
+        non_concise_type: SqlDataType::UNKNOWN_TYPE,
+        searchable: SQL_PRED_BASIC,
+        is_case_sensitive: false,
+        fixed_prec_scale: false,
         scale: None,
         precision: None,
         octet_length: None,
         fixed_bytes_length: None,
+        literal_prefix: None,
+        literal_suffix: None,
+        sql_code: None,
+        is_auto_unique_value: None,
+        is_unsigned: None,
+        num_prec_radix: None,
     };
     pub const BSON: BsonTypeInfo = BsonTypeInfo {
         type_name: "bson",
         sql_type: SqlDataType::UNKNOWN_TYPE,
-        searchable: true,
+        non_concise_type: SqlDataType::UNKNOWN_TYPE,
+        searchable: SQL_PRED_BASIC,
```

Comment:
On closer inspection I am actually a bit unsure _what_ this `bson` type is, unless it is just a default option if no type matches? looking both at the syntax+semantics [types section](https://docs.google.com/document/d/14DnyMSzVNRkoyib8yM62MPJIrkrG1pE0KjRYOWOUuDc/edit#) and the related [bson spec](https://bsonspec.org/spec.html), both line up (in the same order) as the list of datatypes in this file, until we reach `bson` (which does not appear in either specs).

Reply:
Your intuition on this is correct -- the type here that we just call "bson" is not _actually_ a BSON type. We use it instead to represent the "any" type or any mutli-element "anyOf" schema. See [here](https://github.com/mongodb/mongo-odbc-driver/blob/c1b68f647d2364f0a19adf39821ae2bedd283027/core/src/json_schema.rs#L246).

Reply:
^ thanks for checking my intuition on this, that makes sense to me. So then by same token, I think the [spec tests](https://github.com/10gen/mongosql-rs/blob/0cc4cc824a1eeab5cf4fc568bb5fa26a075e2dbc/tests/spec_tests/query_tests/operators_comparison.yml#L306) imply that in some cases, `anyOf` (and thus, this bson type) could be used in a where clause --> so we can keep `SQL_PRED_NONE`. Does that make sense?

Reply:
Hmm, I don't see any `WHERE` clauses in the linked spec tests. But in general, it is possible to have `anyOf` types used in `WHERE` clauses because the only constraint on `WHERE` clauses is that the expression there be of type BOOL or NULL (or MISSING). Any number of expressions/expression compositions can result in that type, such as `a = b` where `a` and `b` have `AnyOf` schemas that are comparable.

> anyOf (and thus, this bson type) could be used in a where clause --> so we can keep SQL_PRED_NONE. Does that make sense?

Part of me thinks if it _can_ be used in a where clause, doesn't that mean it should be one of `SQL_PRED_CHAR`, `SQL_PRED_BASIC`, or `SQL_SEARCHABLE`? (Based on [this documentation](https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlgettypeinfo-function?view=sql-server-ver16) for `SEARCHABLE`). However, I think I agree with you that this should be `SQL_PRED_NONE` because we don't make any guarantees about what this "bson" type _really_ is and there are some types that are truly not comparable in any circumstance, so the safest thing we can say here is `SQL_PRED_NONE`.

tldr: using `SQL_PRED_NONE` for `BsonTypeInfo::BSON`'s `searchable` field sounds good to me 👍